huff_parser/src/lib.rs

```diff
@@ -386,6 +386,43 @@ impl Parser {
         Ok(ConstantDefinition { name, value, span: AstSpan(new_spans) })
     }
 
+    /// Parse a custom error definition.
+    pub fn parse_custom_error(&mut self) -> Result<ErrorDefinition, ParserError> {
+        // Error Identifier
+        self.match_kind(TokenKind::Error)?;
+
+        // Parse the error name
+        self.match_kind(TokenKind::Ident("x".to_string()))?;
+        let tok = self.peek_behind().unwrap().kind;
+        let name = match tok {
+            TokenKind::Ident(err_name) => err_name,
+            _ => {
+                tracing::error!(target: "parser", "TOKEN MISMATCH - EXPECTED IDENT, GOT: {}", tok);
+                let new_spans = self.spans.clone();
+                self.spans = vec![];
+                return Err(ParserError {
+                    kind: ParserErrorKind::UnexpectedType(tok),
+                    hint: Some("Expected error name.".to_string()),
+                    spans: AstSpan(new_spans),
+                })
```

Comment:
We don't have to reset spans here since we return (ala current logic).

So we should be able to _move_ the spans out of the Parser with something like:
```
return Err(ParserError {
    kind: ParserErrorKind::UnexpectedType(tok),
    hint: Some("Expected error name.".to_string()),
    spans: AstSpan(self.spans),
})
```

Though, it might be be beneficial for our future selves.